worlds/lingo/__init__.py

```diff
@@ -89,10 +89,23 @@ def create_items(self):
                     pool.append(self.create_item("Puzzle Skip"))
 
             if traps:
-                traps_list = ["Slowness Trap", "Iceland Trap", "Atbash Trap"]
-
-                for i in range(0, traps):
-                    pool.append(self.create_item(traps_list[i % len(traps_list)]))
+                trap_weights = {name: self.options.trap_weights.get(name, 0) for name in TRAP_ITEMS}
+                total_weight = sum(trap_weights.values())
+
+                if total_weight == 0:
+                    raise Exception("Sum of trap weights must be at least one.")
+
+                trap_counts = {name: int(weight * traps / total_weight) for name, weight in trap_weights.items()}
+                
+                trap_difference = traps - sum(trap_counts.values())
+                if trap_difference > 0:
+                    allowed_traps = [name for name in TRAP_ITEMS if trap_weights[name] > 0]
+                    for i in range(0, trap_difference):
+                        trap_counts[allowed_traps[i % len(allowed_traps)]] += 1
```

Comment:
This could result in having 1 more trap with a lower weight than a trap with a higher weight

Something tells me only I'm pedantic enough to care

Edit: As an example of a strange result, I set the weights to
```
    Atbash Trap: 100
    Iceland Trap: 101
    Slowness Trap: 100
```

And I got 9 Atbash Traps, 11 Iceland Traps and 10 Slowness Traps

Reply:
To care for my world*

Reply:
To be honest I think it's fine 😅